freqtrade/commands/build_config_commands.py

```diff
@@ -107,19 +107,29 @@ def ask_user_config() -> Dict[str, Any]:
             "type": "select",
             "name": "exchange_name",
             "message": "Select exchange",
-            "choices": [
+            "choices": lambda x: [
+                Separator("------ Spot and Perpetual Swaps/Futures -------"),
                 "binance",
+                "gateio",
+                Separator("----------------- Spot Only -------------------"),
```

Comment:
I'd either leave sorting of the exchanges as it was - or have them sorted alphabetically.

This way it looks bad - and if we think about it from a user perspective - (s)he'll already have the exchange in mind (basically you use wherever you have an account) - so this information here is pointless.

Having accounts on most exchanges is something very few people do - so this is just another information that'll be more confusing than helpful.
We will have the "supported" exchanges in the docs - but this will mean that for every new exchange, we'll have to change ordering here - which will fail sometime in the future for sure.

Reply:
People are more likely to choose it if it's at the top, so I sorted them by their level of support

Reply:
Please change sorting to alphabetic.
I think freqtrade should be neutral on this.

we list exchanges we support here for "easy selection" - and each exchange listed will get the same level of support (if something is broken on 2 exchanges, we're not going to fix gate.io, but leave kraken broken for weeks assuming the fix is possible).

What you refer to is functionality (not support) - but even in that case your ordering is wrong - gate.io does not have stoploss orders, nor market orders - so that'd be close to last when it comes to listing.
The sequence is therefore highly subjective (you think about futures to be "the important" feature - but users may not do that.

Reply:
Gateio has stoploss orders on CCXT, is there something in freqtrade that would need to be programmed?

Reply:
Stoploss orders are not unified in ccxt AFAIK - so it'll need to be (slighlty) different code for every exchange.

If you can point me to "how" to do stoploss orders for gateio, i'll surely be adding this (to the develop branch, though).